### PR TITLE
tests: Descriptor wallets, multisig fixes, and general cleanups

### PR DESCRIPTION
**Pull request by achow101**
* Refactor and cleanup wallet, RPC, and `--testnet` argument setup
  * A new wallet is made for each individual test case for now. Funds still provided by the default wallet.
* Multisig setups are refactored and updated to not use the wallet
  * Common functionality is refactored into a separate function to deduplicate a bunch of code.
  * The setup itself is simplified to not require the use of the wallet.
* The multisig descriptor with xpub test added in #353 is moved to its own function
* The multisig `displayaddress` tests for descriptor and paths have been combined as they are basically the same
* The wallet we import stuff into is now a descriptor wallet and we use `importdescriptors` now. This is necessary to fix the Trezor 1 test failure.
* Change multisigs to use the `m/48'` path that is supposedly a standard but not really documented anywhere. This is necessary to fix the Trezor 1 test failure.

### CODE REVIEW

File: test/test_device.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def cleanup_bitcoind():
             pass
 
     # Make sure there are blocks and coins available
-    rpc.generatetoaddress(101, rpc.getnewaddress())
+    rpc.generatetoaddress(150, rpc.getnewaddress())
```
Comment (instagibbs) [inline test/test_device.py:57]: why?

File: test/test_device.py [hunk 2]
```diff
@@ -180,67 +180,60 @@ def setUp(self):
...
-
-        import_result = self.wrpc.importmulti(pkh_keypool_desc)
+        import_result = self.wrpc.importdescriptors(pkh_keypool_desc)
         self.assertTrue(import_result[0]['success'])
         for i in range(0, 21):
-            addr_info = self.wrpc.getaddressinfo(self.wrpc.getnewaddress())
-            self.assertEqual(addr_info['hdkeypath'], "m/44'/1'/0'/0/{}".format(i))
-            addr_info = self.wrpc.getaddressinfo(self.wrpc.getrawchangeaddress())
-            self.assertEqual(addr_info['hdkeypath'], "m/44'/1'/0'/1/{}".format(i))
+            addr_info = self.wrpc.getaddressinfo(self.wrpc.getnewaddress('', 'legacy'))
```
Comment (instagibbs) [inline test/test_device.py:187]: `i` is unused now it seems.

File: test/test_device.py [hunk 3]
```diff
@@ -180,67 +180,60 @@ def setUp(self):
...
+            self.assertTrue(addr_info['hdkeypath'].startswith("m/44'/1'/0'/0/"))
+            addr_info = self.wrpc.getaddressinfo(self.wrpc.getrawchangeaddress('legacy'))
+            self.assertTrue(addr_info['hdkeypath'].startswith("m/44'/1'/0'/1/"))
 
         shwpkh_keypool_desc = self.do_command(self.dev_args + ['getkeypool', '--sh_wpkh', '0', '20'])
-        import_result = self.wrpc.importmulti(shwpkh_keypool_desc)
+        import_result = self.wrpc.importdescriptors(shwpkh_keypool_desc)
         self.assertTrue(import_result[0]['success'])
         for i in range(0, 21):
```
Comment (instagibbs) [inline test/test_device.py:195]: `i` is unused now it seems.

File: test/test_device.py [hunk 4]
```diff
@@ -180,67 +180,60 @@ def setUp(self):
...
+            self.assertTrue(addr_info['hdkeypath'].startswith("m/49'/1'/0'/0/"))
             addr_info = self.wrpc.getaddressinfo(self.wrpc.getrawchangeaddress('p2sh-segwit'))
-            self.assertEqual(addr_info['hdkeypath'], "m/49'/1'/0'/1/{}".format(i))
+            self.assertTrue(addr_info['hdkeypath'].startswith("m/49'/1'/0'/1/"))
 
         wpkh_keypool_desc = self.do_command(self.dev_args + ['getkeypool', '--wpkh', '0', '20'])
-        import_result = self.wrpc.importmulti(wpkh_keypool_desc)
+        import_result = self.wrpc.importdescriptors(wpkh_keypool_desc)
         self.assertTrue(import_result[0]['success'])
         for i in range(0, 21):
```
Comment (instagibbs) [inline test/test_device.py:204]: `i` is unused now it seems.

File: test/test_device.py [hunk 5]
```diff
@@ -180,67 +180,60 @@ def setUp(self):
...
-            self.assertEqual(addr_info['hdkeypath'], "m/49'/1'/3'/0/{}".format(i))
+            self.assertTrue(addr_info['hdkeypath'].startswith("m/49'/1'/3'/0/"))
             addr_info = self.wrpc.getaddressinfo(self.wrpc.getrawchangeaddress('p2sh-segwit'))
-            self.assertEqual(addr_info['hdkeypath'], "m/49'/1'/3'/1/{}".format(i))
+            self.assertTrue(addr_info['hdkeypath'].startswith("m/49'/1'/3'/1/"))
         keypool_desc = self.do_command(self.dev_args + ['getkeypool', '--wpkh', '--account', '3', '0', '20'])
-        import_result = self.wrpc.importmulti(keypool_desc)
+        import_result = self.wrpc.importdescriptors(keypool_desc)
         self.assertTrue(import_result[0]['success'])
         for i in range(0, 21):
```
Comment (instagibbs) [inline test/test_device.py:225]: `i` is unused now it seems.

File: test/test_device.py [hunk 6]
```diff
@@ -180,67 +180,60 @@ def setUp(self):
...
 
         # Test that `--all` option gives the "concatenation" of previous three calls
         all_keypool_desc = self.do_command(self.dev_args + ['getkeypool', '--all', '0', '20'])
         self.assertEqual(all_keypool_desc, pkh_keypool_desc + wpkh_keypool_desc + shwpkh_keypool_desc)
 
         keypool_desc = self.do_command(self.dev_args + ['getkeypool', '--sh_wpkh', '--account', '3', '0', '20'])
-        import_result = self.wrpc.importmulti(keypool_desc)
+        import_result = self.wrpc.importdescriptors(keypool_desc)
         self.assertTrue(import_result[0]['success'])
         for i in range(0, 21):
```
Comment (instagibbs) [inline test/test_device.py:217]: `i` is unused now it seems.

File: test/test_device.py [hunk 7]
```diff
@@ -180,67 +180,60 @@ def setUp(self):
...
+            addr_info = self.wrpc.getaddressinfo(self.wrpc.getnewaddress('', 'bech32'))
+            self.assertTrue(addr_info['hdkeypath'].startswith("m/84'/1'/3'/0/"))
+            addr_info = self.wrpc.getaddressinfo(self.wrpc.getrawchangeaddress('bech32'))
+            self.assertTrue(addr_info['hdkeypath'].startswith("m/84'/1'/3'/1/"))
 
         keypool_desc = self.do_command(self.dev_args + ['getkeypool', '--path', 'm/0h/0h/4h/*', '0', '20'])
-        import_result = self.wrpc.importmulti(keypool_desc)
+        import_result = self.wrpc.importdescriptors(keypool_desc)
         self.assertTrue(import_result[0]['success'])
         for i in range(0, 21):
```
Comment (instagibbs) [inline test/test_device.py:234]: `i` is unused now it seems.

File: test/test_device.py (hunk 1 above)
Comment (achow101) [reply test/test_device.py]: Reverted.

File: test/test_device.py (hunk 2 above)
Comment (achow101) [reply test/test_device.py:187]: Changed to `_` here and elsewhere.

**Comment by instagibbs** [comment]
reACK https://github.com/bitcoin-core/HWI/pull/378/commits/2c381fa2dc4a9d337f42582faf763e27a3fbd885

**Pull request merged**